# HG changeset patch
# User Rik <rik@octave.org>
# Date 1486317619 28800
#      Sun Feb 05 10:00:19 2017 -0800
# Node ID 0d4107676ddf1481560728c388beab25f4dd31e4
# Parent  f81c759dfae7805ea0819f945fc6fd467b67a17e
# Parent  8419d24ab8f705d81f0ff5c4a12961a039da17ca
maint: Periodic merge of stale to default.

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -1873,16 +1873,19 @@ file_editor::construct (void)
   vbox_layout->addWidget (_menu_bar);
   vbox_layout->addWidget (_tool_bar);
   vbox_layout->addWidget (_tab_widget);
   vbox_layout->setMargin (0);
   editor_widget->setLayout (vbox_layout);
   setWidget (editor_widget);
 
   // signals
+  connect (this, SIGNAL (execute_command_in_terminal_signal (const QString&)),
+           main_win (), SLOT (execute_command_in_terminal (const QString&)));
+
   connect (this, SIGNAL (request_settings_dialog (const QString&)),
            main_win (),
            SLOT (process_settings_dialog_request (const QString&)));
 
   connect (main_win (), SIGNAL (new_file_signal (const QString&)),
            this, SLOT (request_new_file (const QString&)));
 
   connect (main_win (), SIGNAL (open_file_signal (const QString&)),
diff --git a/libgui/src/qtinfo/parser.cc b/libgui/src/qtinfo/parser.cc
--- a/libgui/src/qtinfo/parser.cc
+++ b/libgui/src/qtinfo/parser.cc
@@ -574,20 +574,20 @@ replace (QString& text, const QRegExp& r
 }
 
 QString
 parser::global_search (const QString& text, int max_founds)
 {
   QString results;
   QStringList words = text.split (" ",QString::SkipEmptyParts);
 
-  QString re_program ("(" + words.at (0));
+  QString re_program ("(" + QRegExp::escape (words.at (0)));
   for (int i = 1; i < words.size (); i++)
     {
-      re_program += "|" + words.at (i);
+      re_program += "|" + QRegExp::escape (words.at (i));
     }
   re_program += ")";
 
   QRegExp re (re_program, Qt::CaseInsensitive);
 
   results.append ("<html><body>\n<h1>Search results</h1>\n<b>Results for:</b> ");
   results.append (text);
   results.append ("<br>\n");
